sympy/physics/mechanics/particle.py

```diff
@@ -39,56 +41,58 @@ class Particle:
     >>> # Or you could change these later
     >>> pa.mass = m
     >>> pa.point = po
+    >>> # If preferred you can also attach a frame to a Particle
+    >>> pa.add_frame()
+    >>> pa.frame
+    pa_frame
 
     """
+    point = _Body.masscenter
 
-    def __init__(self, name, point, mass):
```

Comment:
Not sure if we would like to add a `frame` kwarg here. Disadvantage would be that the order would be different compared to `RigidBody` due to argument order backwards compatibility. Also if we decide to add the `frame` kwarg here, then we can also add it to `_Body`.

Reply:
Could we add `frame` as a keyword-only argument? I.e. make the initialiser

```python
    def __init__(self, name, point=None, mass=None, *, frame=None):
```

Ideally, I'd suggest we make all arguments except `name` keyword only as that would fix the ordering inconsistency to `RigidBody` issue, but appreciate that that is likely not backwards compatible. In general I prefer keyword-only arguments in instances where there are more than one or two arguments, unless the arguments are so logical that it's not needed (e.g. `x`, `y`, `z` in a 3D vector).

Reply:
```py
Rigidbody('B', Point('Bo'), ReferenceFrame('B'), Symbol('Bm'))
# Decide to make it a Particle
Particle('B', Point('Bo'), ReferenceFrame('B'), Symbol('Bm'))
# if key-word only
raise TypeError: Particle.__init__() takes from 2 to 4 positional arguments but 5 were given
# frame as normal key-word
raise ValueError: Frame must be an instance of ReferenceFrame.
```
Personally think that having it as a normal keyword can give quite a nice error. Yes, this will mean that they do have different order of arguments, but assuming someone uses them as keyword, then this will not pose a problem.

P.S. Will be changing the `frame = ReferenceFrame(frame)` line in the `frame.setter` as throwing a `ValueError` is more sensible.

Reply:
That's poor error messaging on Python's part.

Reply:
Just added the `frame` kwarg.